### PR TITLE
Update for textFitWidth and textFitHeight

### PR DESCRIPTION
**Pull request by DavidBuerer**
Documentation for https://github.com/maplibre/maplibre-gl-js/pull/4019

### CODE REVIEW

File: docs/sprite.md
Comment (HarelM) [inline docs/sprite.md:113]:
```suggestion
   The content rectangle scales to precisely contain the width of its contents but the height will not shrink smaller than the aspect ratio of the original content rectangle. This may be useful scenarios like no. 2 except with vertically written scripts (using `"text-writing-mode": ["vertical"]`).  
```

File: docs/sprite.md
Comment (HarelM) [inline docs/sprite.md:103]:
```suggestion

* stretchOrShrink (or undefined)
```

File: docs/sprite.md
Comment (HarelM) [inline docs/sprite.md:104]:
```suggestion
* `stretchOnly`
```

File: docs/sprite.md
Comment (HarelM) [inline docs/sprite.md:105]:
```suggestion
* `proportional`
```

**Comment by HarelM** [comment]
Thanks for pushing this!!

**Comment by HarelM** [comment]
Can you please add a support table with these two new fields and GL JS version of 4.2.0 please? 
A sdk support table can be copied from deprecations.md file.

**Comment by HarelM** [comment]
@louwers let me know how you would like to track this. 
The linked PRs were discussed heavily in previous monthly meetings, a poll was make to define the best names and implementation in JS is linked too, so this is an addition to the native gaps in terms of implementation.

**Review by louwers — CHANGES_REQUESTED** [review]
Needs a note that these properties are not supported in MapLibre Native.

I can update the documentation when there is a tracking issue.

**Comment by HarelM** [comment]
Yup, this is my comment above about the SDK table support.

**Comment by DavidBuerer** [comment]
I put in an image, relying on GitHub, and it shows up in preview but not in the files list.  Looking at the other image on the page it points back to code repo.  Is that what I should be doing also (a separate check-in to the code repo and then point the docs at it), or is there a new plan going forward?

**Comment by HarelM** [comment]
I think the current image link is OK.
Don't forget the sdk table. 
Looking at the image, it looks like stretchOnly and stretchOrShrink are very similar, is this expected?

**Review by HarelM — APPROVED** [review]

**Comment by DavidBuerer** [comment]
> I think the current image link is OK. Don't forget the sdk table. Looking at the image, it looks like stretchOnly and stretchOrShrink are very similar, is this expected?

Working on the table.  In the diagram, all the dimmed out images are not really meaningful because if you don't set one of the axis to proportional then essentially the system can't engage.  stretchOnly limits the axis to being the aspect ratio of the original sprite or larger.  So the main use case is textFitHeight=proportional, textFitWidth=stretchOnly, and you can see in the table that when the label is narrow, the resulting image has not become too narrow.

**Comment by DavidBuerer** [comment]
> I think the current image link is OK. Don't forget the sdk table. Looking at the image, it looks like stretchOnly and stretchOrShrink are very similar, is this expected?

On the SDK table, I'd rather leave this to someone with more experience with the control.  To get his filled out presumably would require tracking all the way back to beginning and getting all the versions together for all the properties as it would be best to split out `stretchX` and `stretchY` as well?

|SDK Support|MapLibre GL JS|Android SDK|iOS SDK|macOS SDK|
|-----------|--------------|-----------|-------|---------|
|basic functionality| ?? | ?? | ?? | ?? |
|`textFitWidth`, `textFitHeight`| ?? | | | |

**Comment by HarelM** [comment]
You have a point there, I'm not even sure how to get this info...
Not that I think it makes sense to keep track of a feature that predates version 1.x as this version is very old already, but still...
@louwers what do you think we should put in this table?

**Comment by louwers** [comment]
Just put a ✅ if it is supported.

**Comment by DavidBuerer** [comment]
> ✅

Added table with checks, and removed MacOS

**Comment by louwers** [comment]
If it is a new addition I would supply the version. If it is supported in *any* version, I would use the checkmark.

File: docs/sprite.md
Comment (HarelM) [inline docs/sprite.md:72]:
```suggestion
|`textFitWidth`, `textFitHeight`| 4.2.0 | Not supported yet | Not supported yet |
```

**Review by louwers — APPROVED** [review]

**Comment by HarelM** [comment]
I'll merge this after I'll release the new version of gl js.
Thanks @DavidBuerer!

**Pull request merged**
